lib/timeline/component/ItemSet.js

```diff
@@ -1946,13 +1946,34 @@ class ItemSet extends Component {
         fullNestedGroups = fullNestedGroups.concat(nextLevel);
       }
     }
-    let nestedGroups = groupsData.get(fullNestedGroups).map(function (nestedGroup) {
+    var nestedGroups;
+    if(nestingGroup.showNested){
+      var showNestedGroups=groupsData.get(nestingGroup.nestedGroups);
+        let i=0;
+        while (i<showNestedGroups.length){
+          let group=showNestedGroups[i];
+          if(group.nestedGroups && group.nestedGroups.length>0 && (group.showNested == undefined || group.showNested == true)){
+            showNestedGroups.push(...groupsData.get(group.nestedGroups));
+          }
+          i=i+1;
+        }						
+           nestedGroups = showNestedGroups.map(function (nestedGroup) {
+          if (nestedGroup.visible == undefined) {
+            nestedGroup.visible = true;
+          }			
+          nestedGroup.visible = !!nestingGroup.showNested;
+
+          return nestedGroup;
+          });
+    }
+    else{
+     nestedGroups = groupsData.get(fullNestedGroups).map(function (nestedGroup) {
         if (nestedGroup.visible == undefined) {
           nestedGroup.visible = true;
         }
         nestedGroup.visible = !!nestingGroup.showNested;
         return nestedGroup;
-      });
+      });}
```

Comment:
please fix the code formatting to fit to the rest of the code-base.

Reply:
Done using 'prettier'.